File: python/1070_permutations.py
import itertools

n = int(input())

if n == 1:
    print(1)
elif n < 4:
    print("NO SOLUTION")
else:
    print(
        " ".join(
            str(x) for x in itertools.chain(range(2, n + 1, 2), range(1, n + 1, 2))
        )
    )

File: src/1070_permutations.rs
use std::io;

fn main() -> Result<(), Box<dyn std::error::Error>> {
    let mut buff = String::new();

    io::stdin().read_line(&mut buff)?;
    let n: usize = buff.trim().parse()?;

    match n {
        1 => println!("1"),
        2 | 3 => println!("NO SOLUTION"),
        n => (2..=n)
            .step_by(2)
            .chain((1..=n).step_by(2))
            .for_each(|x| print!("{} ", x)),
    }
    Ok(())
}
